app/models/tweet.rb

```diff
@@ -0,0 +1,51 @@
+class Tweet
+  attr_reader :id, :user_id, :title, :tweet_content,  :errors
+  require_relative 'user'
+
+  def initialize(id:, user_id:, title:, tweet_content:)
+    @id = id
+    @user_id = user_id
+    @title = title
+    @tweet_content = tweet_content
+    @errors = []
+  end
+
+  def present?
+    @id != nil
+  end
+
+  def self.build_by_user(user:, id:, title:, tweet_content:)
+    user_id = user.id
+    Tweet.new(id: id, user_id: user_id, title: title, tweet_content: tweet_content)
+  end
+
+  def title_valid?
+    if @title == nil || @title.empty?
+      @errors << 'titleを入力して下さい。'
+      return false
+    elsif @title.length > 15
+      @errors << 'titleは15文字以内にして下さい。'
+      return false
+    else
+      return true
+    end
+  end
+
+  def tweet_content_valid?
+    if @tweet_content == nil || @tweet_content.empty?
+      @errors << 'tweet_contentを入力して下さい。'
+      return false
+    elsif @tweet_content.length > 100
+      @errors << 'tweet_contentは100文字以内にして下さい。'
+      return false
+    else
+      return true
+    end
+  end
+
+  def valid?
+    result_title_valid = title_valid?
+    result_tweet_content_valid = tweet_content_valid?
+    result_title_valid && result_tweet_content_valid
```

Comment:
Simple

こちらもわざわざ一時変数に格納する必要がないと思う！
```suggestion
    title_valid? && tweet_content_valid?
```

Reply:
@ochi-sho-private-study 
` titile_valid? && tweet_content_valid?` にすると、` title_valid?` がfalseのときに` tweet_content_valid?` が評価されない
どちらもfalseになるとき、両方のエラーメッセージをerrorに格納するために、一時変数にしてる
エラーメッセージが一つしか入らないなら、errorを配列にしなくていいだろうし

Reply:
```ruby
def invalid?
  title_invalid? && tweet_content_invalid?
end

def title_invalid?
  # titleがvalidの時false, titleがinvalidの時はtrueにしてエラーを格納する。
end

def tweet_content_invalid?
  # tweet_contentがvalidの時false, tweet_contentがinvalidの時はtrueにしてエラーを格納する。
end
```